<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<script type="text/javascript" src="design-project.js"></script>
<link rel="stylesheet" type="text/css" href="design-project.css">
</head>

<body>
<div id="wrapper">
	<form id="form1">
		<div id="formLeft">
		
			Step 1. Give a name to this project:<br>
			<input type="text" name="projectName" value="My project" />
			<div id="projectNameValidationInfo" class="validationMessage"></div>
			<br><br>
			Step 2. Choose your analysis setup:
			<br><br>
			Mapping-by-sequencing strategy:<br>
			<div>
				<div class="mx-button">
					<input type="radio" class="analysisType" name="mx12" id="button1" />
					<label for="button1" unselectable>Tagged sequence mapping</label>
				</div>
				<div class="mx-button">
					<input type="radio" class="analysisType" name="mx12" id="button2" />
					<label for="button2" unselectable>Linkage analysis mapping</label>
				</div>
			</div>
		
			<br><br><br>
		
			Data source:<br>
			<div>
				<div class="mx-button">
					<input type="radio" class="dataSource" name="mx34" id="button3" />
					<label for="button3" unselectable>Use my own data</label>
				</div>
				<div class="mx-button">
					<input type="radio" class="dataSource" name="mx34" id="button4" />
					<label for="button4" unselectable>Simulate data</label>
				</div>
			</div>
		
			<br><br><br>
		
			Type of NGS library:<br>
			<div>
				<div class="mx-button">
					<input type="radio" class="libType" name="mx56" id="button5" />
					<label for="button5" unselectable>Single end</label>
				</div>
				<div class="mx-button">
					<input type="radio" class="libType" name="mx56" id="button6" />
					<label for="button6" unselectable>Paired end</label>
				</div>
			</div>
		
			<br><br><br>
		
			Genome sequence:<br>
			<select multiple id="refSeqs">
				<option value="chr1">Chr1</option>
				<option value="chr2">Chr2</option>
				<option value="chr3">Chr3</option>
				<option value="chr4">Chr4</option>
			</select>
		
			<br>
			
			<div id="insSeqSelector">
				<select id="insSeq">
					<option value="default">Choose insertion sequence file</option>
					<option value="insertion.fa">insertion.fa</option>
					<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
					<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
					<option value="reads_f.fq">reads_f.fq</option>
					<option value="reads_r.fq">reads_r.fq</option>
					<option value="reads_se.fq">reads_se.fq</option>
				</select>
			</div>
		
			<br>
		
			<select id="gffFile">
				<option value="default">Choose GFF3 file</option>
				<option value="insertion.fa">insertion.fa</option>
				<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
				<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
				<option value="reads_f.fq">reads_f.fq</option>
				<option value="reads_r.fq">reads_r.fq</option>
				<option value="reads_se.fq">reads_se.fq</option>
			</select>
		
			<br>
		
			<select id="annFile">
				<option value="default">Choose gene annotation file</option>
				<option value="insertion.fa">insertion.fa</option>
				<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
				<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
				<option value="reads_f.fq">reads_f.fq</option>
				<option value="reads_r.fq">reads_r.fq</option>
				<option value="reads_se.fq">reads_se.fq</option>
			</select>
		</div>

		<div id="formRight">
		
			<!--
				All the divs inside <div id="formRight"> are hidden until the user
				interacts with buttons in <div id="formLeft">
			-->
			
			<div id="expDataInterface">
				Experimental data chosen
							
				<div id="expDataIns">
					<div id="expDataSingle">
						<select id="readsSingle">
							<option value="default">Choose single-end reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
					</div>
					
					<div id="expDataPaired">
						<select id="readsForward">
							<option value="default">Choose forward reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
						<select id="readsReverse">
							<option value="default">Choose reverse reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
					</div>
				</div>
					
				<div id="expDataSnp">
					
					<div id="expDataSnpInfo">				
						<div>
							Mutant strain<br>
							<input type="radio" id="refStrainYes" name="isRefStrain">
							<label for="refStrainYes">The same as reference strain</label>
							<br>
							<input type="radio" id="refStrainNo" name="isRefStrain">
							<label for="refStrainNo">Other than the reference strain</label>
						</div>
						<br>
						<div>
							Type of mapping cross performed<br>
							<input type="radio" id="crossTypeBack" name="crossType">
							<label for="crossTypeBack">Backcross</label>
							<br>
							<input type="radio" id="crossTypeOut" name="crossType">
							<label for="crossTypeOut">Outcross</label>
						</div>
						<br>
						<div>
							Parental reads provided<br>
							<input type="radio" id="parentalReadsMutant" name="parentalReads">
							<label for="parentalReadsMutant">The mutant's</label>
							<br>
							<input type="radio" id="parentalReadsNoMutant" name="parentalReads">
							<label for="parentalReadsNoMutant">The other</label>
						</div>
					</div>
					
					<div id="expDataSingleTwosamples">
						<select id="readsSinglePar">
							<option value="default">Parental - choose reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
						<br>
						<select id="readsSingleF2">
							<option value="default">F2 - choose reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
					</div>
				
					<div id="expDataPairedTwosamples">
						<select id="readsForwardPar">
							<option value="default">Parental - choose forward reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
						<select id="readsReversePar">
							<option value="default">Parental - choose reverse reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
						<br>
						<select id="readsReversePar">
							<option value="default">F2 - choose forward reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
						<select id="readsReverseF2">
							<option value="default">F2 - choose reverse reads</option>
							<option value="insertion.fa">insertion.fa</option>
							<option value="arabidopsis_gff.gff">arabidopsis_gff.gff</option>
							<option value="at-gene-funct-tair10.txt">at-gene-funct-tair10.txt</option>
							<option value="reads_f.fq">reads_f.fq</option>
							<option value="reads_r.fq">reads_r.fq</option>
							<option value="reads_se.fq">reads_se.fq</option>
						</select>
					</div>
				</div>
			</div>
			
			<div id="simDataInterface">
				Simulated data chosen
				<div id="simDataIns">				
					<div id="simMutInterface">
						Number of mutations:<br>
						<input type="text" name="simMutNbr" class="simNumericInput" value="" />
						<br><br>
						Mode:<br>
						<input type="radio" name="simMutMode" id="simMutModeD">
						<label for="simMutModeD">Drift</label>
						<br>
						<input type="radio" name="simMutMode" id="simMutModeE">
						<label for="simMutModeE">EMS (GC > AT)</label>
						<br>
						<input type="radio" name="simMutMode" id="simMutModeL">
						<label for="simMutModeL">Large DNA insertions</label>
					</div>
				
					<div id="simRecselInterface">
						<br><br>No RecSel here<br><br>
					</div>
				
					<div id="simSeqInterface">
						Read depth (X):<br>
						<input type="text" name="simSeqRD" class="simNumericInput" value="40" />
						<br>
						Read length [mean, SD] (nt):<br>
						<input type="text" name="simSeqRL" class="simNumericInput" value="90,0" />
						<br>
						<div id="simSeqFL">
							Fragment length [mean, SD] (nt):<br>
							<input type="text" name="simSeqFL" class="simNumericInput" value="500,100" />
						</div>
						Basecalling error rate (%):<br>
						<input type="text" name="simSeqBER" class="simNumericInput" value="1" />
						<br>
						GC bias strength (%):<br>
						<input type="text" name="simSeqGBS" class="simNumericInput" value="50" />
						<br>
					</div>
				
				</div>
				<div id="simDataSnp">
				
					Simulate Mapping by Sequencing data
				
				</div>
				
			</div>
			
		</div>
	
		<div id="formButtons">
			<input type="button" class="button" id="runButton" value="Run analysis" /> 
			<input type="button" class="button" id="otherButton" value="Other button" />
		</div>
		
	</form>
	
	<div id="command">
		<p>
			$project_name $workflow $data_source $lib_type $ref_seq $ins_seq $read_s
			$reads_f $reads_r $gff_file $ann_file $sim-mut $sim-recsel $sim-seq
		</p>
		<p id="commandString"></p>
	</div>
<div>
</body>
</html>
